lib/Group/GroupBackend.php

```diff
@@ -0,0 +1,150 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Workspace\Group;
+
+use OCA\Workspace\Service\Group\ConnectedGroupsService;
+use OCP\Group\Backend\ABackend;
+use OCP\GroupInterface;
+use OCP\IGroupManager;
+use OCP\IUserManager;
+
+class GroupBackend extends ABackend implements GroupInterface {
+
+	private bool $_avoidRecurse_users;
+	private bool $_avoidRecurse_groups;
+
+	/**
+	 * @param IGroupManager $groupManager parent group manager
+	 */
+	public function __construct(protected IGroupManager $groupManager, protected IUserManager $userManager, private ConnectedGroupsService $connectedGroups) {
+		$this->_avoidRecurse_users = $this->_avoidRecurse_groups = false;
+	}
+
+	/**
+	 * is user in group?
+	 * @param string $uid uid of the user
+	 * @param string $gid gid of the group
+	 * @return bool
+	 * @since 4.5.0
+	 *
+	 * Checks whether the user is member of a group or not.
+	 */
+	public function inGroup($uid, $gid) {
+		return true;
+	}
+
+	/**
+	 * Get all groups a user belongs to
+	 * @param string $uid Name of the user
+	 * @return array an array of group names
+	 * @since 4.5.0
+	 *
+	 * This function fetches all groups a user belongs to. It does not check
+	 * if the user exists at all.
+	 */
+	public function getUserGroups($uid) {
+		if ($this->_avoidRecurse_groups) {
+			return [];
+		}
+
+		$avoid = $this->_avoidRecurse_groups;
+		$this->_avoidRecurse_groups = true;
+		$user = $this->userManager->get($uid);
+		if ($user) {
+			$groupIds = $this->groupManager->getUserGroupIds($user);
+		} else {
+			$groupIds = [];
+		}
+		$this->_avoidRecurse_groups = $avoid;
+		if (empty($groupIds)) {
+			return [];
+		}
+		$userGroups = [];
+		foreach($groupIds as $gid) {
+			$connectedGids = $this->connectedGroups->getConnectedSpaceToGroupIds($gid);
+			if ($connectedGids !== null) {
+				$userGroups = array_merge($userGroups, $connectedGids);
+			}
+		}
+
+		return $userGroups;
+	}
+
+	/**
+	 * get a list of all groups
+	 * @param string $search
+	 * @param int $limit
+	 * @param int $offset
+	 * @return array an array of group names
+	 * @since 4.5.0
+	 *
+	 * Returns a list with all groups
+	 */
+	public function getGroups($search = '', $limit = -1, $offset = 0) {
+		return []; // return all virtual groups
+	}
+	
+	/**
+	 * check if a group exists
+	 * @param string $gid
+	 * @return bool
+	 * @since 4.5.0
+	 */
+	public function groupExists($gid) {
+		return true;
+	}
+
+	/**
+	 * get a list of all users in a group
+	 * @param string $gid
+	 * @param string $search
+	 * @param int $limit
+	 * @param int $offset
+	 * @return array an array of user ids
+	 * @since 4.5.0
+	 */
+	public function usersInGroup($gid, $search = '', $limit = -1, $offset = 0) {
```

Comment:
Could you type your parameters, please? :)

Reply:
refused by my syntax linter, as the interface does not